### DISCUSSION TITLE
Finding out which stems are in which models

### DISCUSSION THREAD

**Discussion by sanderness66** [nomadkaraoke/python-audio-separator#130] [Q&A] [answered]
Hello! Am I right in thinking that only the demucs models separate into drums/bass/vocal/other stems and the others mostly into vocals/instruments? How can I find out which model separates in which stems?

**Comment by beveradb** [comment, marked as answer]
Sorta; there are a lot of models with different purposes, but the majority are two stem (instrumental and vocals) models.
Usually the name of the model will give you an idea of its purpose.

Run `audio-separator -l` and read through the list of models;
if it's not obvious from the name, it's probably a 2-stem (Vocals and Instrumentals) model.

Here are some examples of models which are not just vocals/instrumental:
- `htdemucs_6s.yaml` tries to separate 6 stems from the input audio (Vocals, Bass, Drums, Guitar, Piano, Other)

- `mel_band_roformer_crowd_aufr33_viperx_sdr_8.7144.ckpt` tries to separate crowd noise
- `denoise_mel_band_roformer_aufr33_sdr_27.9959.ckpt` and `UVR-DeNoise` try to remove background noise
- `UVR-De-Echo-Aggressive.pth` tries to remove echo
- `deverb_bs_roformer_8_384dim_10depth.ckpt` and `Reverb_HQ_By_FoxJoy.onnx` try to remove reverb

- `UVR-BVE-4B_SN-44100-1` is designed to separate background vocals (BVE is "Backing Vocals Extraction")
- Any with `Karaoke` in the name e.g. `6_HP-Karaoke-UVR`, `Mel-Roformer-Karaoke-Aufr33-Viperx`, `UVR-MDX-NET Karaoke` etc. will be a 2-stem model but if you give it only a vocal input it will output Lead Vocals and Backing Vocals separately.

- `17_HP-Wind_Inst-UVR.pth` tries to separate Wind instruments specifically
- The various `kuielab_` models are all 2 stem models but focus on a specific stem, e.g.
```
        "MDX-Net Model: kuielab_a_bass": "kuielab_a_bass.onnx",
        "MDX-Net Model: kuielab_a_drums": "kuielab_a_drums.onnx",
        "MDX-Net Model: kuielab_a_other": "kuielab_a_other.onnx",
        "MDX-Net Model: kuielab_a_vocals": "kuielab_a_vocals.onnx",
```

Hope this gives you an idea; overall though just experiment with different models till you get the result you want!

**Reply by sanderness66** [reply]
Thanks! There are some models in there that sound rather interesting!

**Comment by beveradb** [comment]
FYI I've now added a more comprehensive post explaining different use cases and recommending models:

https://github.com/nomadkaraoke/python-audio-separator/discussions/133

**Reply by sanderness66** [reply]
Very useful, thanks!
